spec/workers/web/push_notification_worker_spec.rb

```diff
@@ -37,17 +37,31 @@
 
       allow(Webpush::Encryption).to receive(:encrypt).and_return(payload)
       allow(JWT).to receive(:encode).and_return('jwt.encoded.payload')
-
-      stub_request(:post, endpoint).to_return(status: 201, body: '')
     end
 
     it 'calls the relevant service with the correct headers' do
+      stub_request(:post, endpoint).to_return(status: 201, body: '')
+
       subject.perform(subscription.id, notification.id)
 
       expect(web_push_endpoint_request)
         .to have_been_made
     end
 
+    it 'raises when request fails' do
+      stub_request(:post, endpoint).to_return(status: 500)
+
+      expect { subject.perform(subscription.id, notification.id) }.to raise_error Mastodon::UnexpectedResponseError
+    end
```

Comment:
It feels like we should have some tests for the "stoplight did it's thing" but I don't see any example of those in the repository.